Description: fix for mkdirp 1
Author: Xavier Guimard <yadd@debian.org>
Forwarded: https://github.com/jprichardson/node-klaw/issues/40
Last-Update: 2020-10-23

--- a/tests/_test.js
+++ b/tests/_test.js
@@ -10,17 +10,16 @@
     var testDir = path.join(os.tmpdir(), 'klaw-tests')
     rimraf(testDir, function (err) {
       if (err) return t.end(err)
-      mkdirp(testDir, function (err) {
-        if (err) return t.end(err)
-
+      mkdirp(testDir).then(() => {
         var oldEnd = t.end
         t.end = function () {
           rimraf(testDir, function (err) {
             err ? oldEnd.apply(t, [err]) : oldEnd.apply(t, arguments)
           })
         }
-
         testFn(t, testDir)
+      }).catch((err) => {
+        return t.end(err)
       })
     })
   })
